lib/transproc.rb

```diff
@@ -18,6 +19,7 @@ module Transproc
   # @api public
   def register(*args, &block)
     name, fn = *args
+    raise Error.new("function #{name} is already defined") if functions.include?(name)
```

Comment:
It would be better to have a custom error for this. ie `FunctionAlreadyRegisteredError`.